### ISSUE TITLE
Configuration is not unique per Tap when doing a hashJoin

### ISSUE THREAD

**Issue by colinmarc** [twitter/scalding#1262]
Hi!

I ran into a case recently where I was using parquet projection on one source in a hash join, but not the other. Parquet uses a key in the hadoop configuration to specify the projection string when reading:

https://github.com/apache/incubator-parquet-mr/blob/master/parquet-cascading/src/main/java/parquet/cascading/ParquetValueScheme.java#L108-L112

That means that if it's set in the global configuration for a job, and multiple sources are loaded, sources could be loaded with the wrong projection string, which could cause all sorts of issues (but mostly, errors). Fortunately, Cascading forks the config at some point, passing an instance into your `Scheme` that is specific to that tap.

However, when `hashJoin`ing, this appears to not be the case. When I set a projection string for one source (the left side of the join), but not the right, the projection string is used for both, which causes the job to crash. This problem goes away if I set a projection string for both sources, or neither.

**Comment by ianoc** [comment]
What sort of error occurs? just that its missmatched in the projections?

**Comment by johnynek** [comment]
paging @cwensel

**Comment by colinmarc** [comment]
Yeah, in this case it was just mismatched projections. You could imagine a case where the sources have similar fields and the projection gets applied where it shouldn't, though. And who knows what other source-specific config depends on this not happening.

**Comment by cwensel** [comment]
Could be many things. Would love a test case.

https://github.com/cwensel/cascading/blob/wip-2.6/CONTRIBUTING.md
